dateparser/utils.py

```diff
@@ -86,3 +87,37 @@ def find_date_separator(format):
     m = re.search(r'(?:(?:%[dbBmaA])(\W))+', format)
     if m:
         return m.group(1)
+
+
+class Registry(object):
+
+    _global_dict = {}
+    _tag = 'from_registry'
+
+    def __new__(cls, *args, **kwargs):
+        key = cls.get_key(*args, **kwargs)
+
+        if key in cls._global_dict:
+            return cls._tag_instance(cls._global_dict[key])
+
+        return cls._global_dict.setdefault(
+            key,
+            object.__new__(cls, *args, **kwargs)
+        )
```

Comment:
whenever we return an instance from _global_dict, we tag it with `from_register` attribute so that `__init__` doesn't get applied to it, The suggested structure would interfere with this condition.